File: src/main/java/com/github/thorben/webnd/domain/character/DetailType.java
package com.github.thorben.webnd.domain.character;

import com.github.thorben.webnd.domain.games.GameType;
import com.github.thorben.webnd.foundation.AbstractEntity;

import javax.persistence.*;

@Entity(name = "DetailsType")
@Table(name = "detail_types")
public class DetailType extends AbstractEntity<Integer> {

	@Id
	@GeneratedValue
	private Integer id;
	@OneToOne
	private GameType gameType;
	@Column(name = "detail_type_name", unique = true)
	private String name;
	@Column(name = "detail_type_description")
	private String description;
	@Column(name = "detail_type__required")
	private boolean required;

	protected DetailType() {
		// JPA
	}

	public DetailType(GameType gameType, String name, String description) {
		this.gameType = gameType;
		this.name = name;
		this.description = description;
	}

	@Override
	public Integer getId() {
		return id;
	}

	public GameType getGameType() {
		return gameType;
	}

	public String getName() {
		return name;
	}

	public String getDescription() {
		return description;
	}

	public boolean isRequired() {
		return required;
	}

	public void setRequired(boolean required) {
		this.required = required;
	}
}

File: src/main/kotlin/com/github/thorben/webnd/domain/character/DetailType.kt
package com.github.thorben.webnd.domain.character

import com.github.thorben.webnd.domain.games.GameType
import com.github.thorben.webnd.foundation.AbstractEntity
import javax.persistence.*

@Entity(name = "DetailsType")
@Table(name = "detail_types")
class DetailType : AbstractEntity<Int?> {
	@Id
	@GeneratedValue
	override val id: Int? = null

	@ManyToOne
	var gameType: GameType? = null
		private set

	@Column(name = "detail_type_name", unique = true)
	var name: String? = null
		private set

	@Column(name = "detail_type_description")
	var description: String? = null
		private set

	@Column(name = "detail_type__required")
	var isRequired = false

	protected constructor() {
		// JPA
	}

	constructor(gameType: GameType?, name: String?, description: String?) {
		this.gameType = gameType
		this.name = name
		this.description = description
	}

}